lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, :address, :phone, :services, :registered_vehicles, :collected_fees, :hours, :closed_holidays
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_values)
+    @name = facility_values[:name]
+    @address = facility_values[:address]
+    @phone = facility_values[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @hours = nil
+    @closed_holidays = nil
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      if vehicle.antique?
+        @collected_fees += 25
+        vehicle.plate_type = :antique
+      elsif vehicle.electric_vehicle?
+        @collected_fees += 200
+        vehicle.plate_type = :ev
+      else
+        @collected_fees += 100
+        vehicle.plate_type = :regular
+      end
+    else
+      "Error: #{name} does not offer Vehicle Registration services."
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?("New Drivers License")
+      if !registrant.permit? || registrant.age < 16
+        "Error: Written tests can only be administered to registrants with a permit, and who are at least 16 years of age."
+      else
+        registrant.license_data[:written] = true
+      end
+    else
+      "Error: #{@name} does not offer Written Test services."
+    end
```

Comment:
Same thing here. Could use more logical operators to flatten this out. But overall, great job with the logic.